File: sdks/python-sdk/agentlabs/attachment.py
import os
from typing import TypedDict, Optional

MAX_ATTACHMENT_SIZE = 1024 * 1024 * 10

class AttachmentItem:
    def __init__(self):
        pass

class _UploadedAttachment(TypedDict):
    id: str

class Attachment:
    @staticmethod
    def from_local_file(path: str, mime_type: Optional[str] = None) -> AttachmentItem:
        return _LocalFileAttachment(path, mime_type)

    @staticmethod
    def from_buffer(data: bytes, filename: str, mime_type: Optional[str] = None) -> AttachmentItem:
        return _BufferFileAttachment(data, filename, mime_type)

class _LocalFileAttachment(AttachmentItem):
    data: bytes

    def __init__(self, path: str, mime_type: Optional[str]):
        self.path = path
        self.filename = os.path.basename(path)
        self.mime_type = mime_type

    def load(self):
        file = open(self.path, 'rb')
        data = file.read()

        if len(data) > MAX_ATTACHMENT_SIZE:
            raise Exception("Attachment too big, max size is 10MB")

        self.data = data

class _BufferFileAttachment(AttachmentItem):
    data: bytes

    def __init__(self, data: bytes, filename: str, mime_type: Optional[str]):
        self.filename = filename
        self.data = data
        self.mime_type = mime_type
    
    def load(self):
        if len(self.data) > MAX_ATTACHMENT_SIZE:
            raise Exception("Attachment too big, max size is 10MB")

        pass

File: sdks/node-sdk/src/attachment.ts
import { readFile } from "fs/promises";
import { basename } from 'path';

export class Attachment {
	static fromLocalFile(filePath: string, options: FileAttachmentOptions = {}): AttachmentItem {
		return new LocalFileAttachment(filePath, options);
	}

	/**
	 * Creates an attachment from in-memory data held in a Buffer.
	 * You need to specify the filename manually as there is no file path to infer it from.
	 */
	static fromBuffer(buffer: Buffer, filename: string, options: FileAttachmentOptions = {}): AttachmentItem {
		return new BufferFileAttachment(buffer, filename, options);
	}
}

export abstract class AttachmentItem {}

export interface FileAttachmentOptions {
	/**
	 * A custom mime type to enforce on the attachment.
	 * Unless the file has an unusual file extension, you are often better
	 * off letting the mime type be detected automatically by the server.
	 */
	mimeType?: string;
}

export class LocalFileAttachment extends AttachmentItem {
  static maxFileSize = 10 * 1024 * 1024; // 10 MB
  public buffer = Buffer.alloc(0);
  public filename: string;

  constructor(public filePath: string, public options: FileAttachmentOptions = {}) {
	super();
	this.filename = basename(filePath);
  }

  async load(): Promise<void> {
	  this.buffer = await readFile(this.filePath);

	  if (this.buffer.length > LocalFileAttachment.maxFileSize) {
		  throw new Error(`File size exceeds 10MB, which is beyond the maximum allowed size for a single attachment. Consider splitting them up if possible.`);
	  }
  }
}

export class BufferFileAttachment extends AttachmentItem {
  static maxFileSize = 10 * 1024 * 1024;

  constructor(public buffer: Buffer, public filename: string, public options: FileAttachmentOptions = {}) {
	super();
  }

  async load(): Promise<void> {
	  if (this.buffer.length > BufferFileAttachment.maxFileSize) {
		  throw new Error(`File size exceeds 10MB, which is beyond the maximum allowed size for a single attachment. Consider splitting them up if possible.`);
	  }
  }
}



